rj_gameplay/rj_gameplay/tactic/goalie_tactic.py

```diff
@@ -103,7 +105,82 @@ def get_requests(self, world_state: rc.WorldState,
         :return: A list of role requests for move skills needed
         """
 
+        def track_utility() -> float:
+            """
+            :return: the utility of track action
+            """
+            if ball_speed == 0:
+                ball_to_goal_time = 100
+            else:
+                ball_to_goal_time = ball_dist / ball_speed
+            return ball_to_goal_time / 100
+
+        def receive_utility() -> float:
+            """
+            :return: the utility of receive action
+            """
+            # this probably makes sense to stay hard coded like this
+            # but maybe there is a better option?
+            if ball_speed < .5 and (
+                abs(ball_pos[0]) < box_w / 2 + line_w + MAX_OOB
+                and ball_pos[1] < box_h + line_w + MAX_OOB
+            ) and not world_state.game_info.is_stopped():
+                return 1
+            else:
+                return 0
+
+        def block_utility() -> float:
+            """Increases as ball to goal time approaches 0.
+            :return: the utility of block action
+            """
+            if ball_speed > 0 and np.dot(towards_goal,
+                                         world_state.ball.vel) > 0.3:
+                # if ball is moving and coming at goal, move laterally to block ball
+                ball_to_goal_time = ball_dist / ball_speed
+            else:
+                ball_to_goal_time = 100
+            return 1 - (ball_to_goal_time / 100)
+
+        def clear_utility() -> float:
+            """
+            :return: the utility of clear action
+            """
+            # should clear if no open pass available
+            # how do I check that?
+            return 0
```

Comment:
Our passing isn't working right now, leave it as is for now.

Reply:
Actually, set the value to 1 for "always clear" for now until we get passing.
```suggestion
            return 1
```